gems/smithy/lib/smithy/tools/underscore.rb

```diff
@@ -20,6 +20,14 @@ def underscore(string)
           new_string.downcase!
           new_string
         end
+
+        # @param [String<JMESPath>]
+        # @return [String]
+        def underscore_jmespath(expression)
```

Comment:
Is this taken from somewhere?

Reply:
Thinking from V3 codegen: https://github.com/aws/aws-sdk-ruby/blob/version-3/build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/underscore.rb#L23-L29